github_workflow.md

```diff
@@ -0,0 +1,161 @@
+  committed to implementing. Essentially, `Backlog` means "at some point in the
+  future, maybe."
+* Subprojects should have a `No Action` milestone to indicate issues or pull
+  requests that were closed with no action or work being performed. In general
+  we try to close issues/PRs with this milestone.
+* The only time an issue should have no milestone is early on in the triage
+  process when it still has the `status:Needs Info` label.
+
+## Labels
+
+We use GitHub labels to organize issues and pull requests
+(https://github.com/jupyterlab/jupyterlab/labels). The Github issues namespace
+is completely unstructured; each label is simply a name with an associated
+color. This unstructured nature, makes it difficult to organize work
+effectively. To address, this challenge, we are started to use a more formal
+label schema/taxonomy.
+
+### Internal-facing Labels
+
+Typically, every label should have one of four prefixes:
+
+[1. `type:`](`type:`)
+[2. `component:`](`component:`)
+[3. `cat:`](`cat:`)
+[4. `status:`](`status:`)
+
+All issues of a given prefix should have the same color to make visual
+identification easy.
+
+#### `type:`
+
+The `type:` prefix indicates the type of issue or pull request. There are four
+types:
+
+* `type:bug`
+* `type:enhancement`
+* `type:maintenance`
+* `type:question`
+
+Guidelines:
+
+* Each issue or pull request should have only one `type:` prefixed label.
+* Limiting ourselves to these four types allows us to query our issues in a 
+  structural and relational manner.
+
+#### `component:`
+
+The `component:` prefix indicates the component or part of the software the
+issue is related to. Thus, on the JupyterLab repository, we have labels such as:
+
+* `component:About`
+* `component:Application`
+* `component:Build System`
+* `component:Command Palette`
+* `component:Completer`
+* `component:Console`
+* `component:Notebook`
+* `component:Terminal`
+* `component:File Browser`
+* `component:File Editor`
+* ...
+
+Guidelines:
+
+* A single issue or pull request can have multiple `component:` labels, but
+  usually it will be a small number.
+* Each subproject will have its own unique set of `component:` labels.
+
+#### `cat:`
+
+The `cat:` prefix indicates the "category" of the issue. The category is sort of
+mixin to the `component:` label to indicate cross-cutting categories such as:
+
+* `cat:Accesibility`
+* `cat:Design and UX`
+* `cat:Documentation`
+* `cat:Examples`
+* `cat:Settings`
+
+Guidelines:
+
+* A single issue or pull request can have multiple `cat:` labels.
+* Each subproject will have its own unique set of `cat:` labels, but effort
+  should be taken to use common names.
+
+#### `status:`
+
+The `status:` prefix indicates the status of the issue or pull request, such as:
+
+* `status:Blocked`
+* `status:Duplicate`
+* `status:Inactive`
+* `status:Needs Discussion`
+* `status:Needs Info`
+* `status:Needs Review`
+* `status:Needs User Testing`
+* `status:Wont Fix`
```

Comment:
We should remove this as it is not contributor friendly.

Reply:
Just the `wont-fix` label @ellisonbg 👍

Reply:
I really like just putting those won't fix issues under a Reference milestone - they're there for future reference of the discussions, but no need to do anything. Anything that has a specific work item to do is under a Future or specific version milestone.

And using a milestone is a good way to classify "won't fix" issues instead of a label, since "won't fix" is a resolution, not an intermediate status. (I suppose the "duplicate" and "inactive" are essentially comments on why something is being resolved as well - I'd probably say keep the duplicate label, get rid of the inactive label, and just classify duplicate, inactive, and won't fix under a Reference milestone, and leave it to the comments to be more nuanced about whether it's a won't fix, or if the discussion trailed off and the ticket is inactive...)

Reply:
(Edit: put my comments as replies to Carol's so they make sense in this long, multipart threaded discussion...)

Reply:
Given the reasoning in Ian Bicking's blog post, I'd also be in favor of getting rid of the duplicate label, and just put more nuanced text as a comment. I like the idea of not worrying too much about fine-grained classification with labels, but instead have broad categories that make it easy to determine what should be done, and leave more nuanced sentiments to comments and interaction with the people.

Reply:
The one benefit of keeping `duplicate` is that it can be used as a heads up that we have some tweaking to do on documentation if the same issue keeps coming up.

Reply:
Sure, that makes sense too.

Reply:
I'm also in favor of those changes